### PR TITLE
s3: handle 307 redirects

### PR DESCRIPTION
**Pull request by alxndrsn**
Depends on https://github.com/getodk/central/pull/715

### CODE REVIEW

File: files/nginx/odk.conf.template [hunk 1]
```diff
@@ -58,6 +68,22 @@ server {
     proxy_request_buffering on;
     proxy_buffering off;
     proxy_read_timeout 2m;
+
+    # transparently follow 307 redirects
+    # See: https://serverfault.com/a/792035
+    proxy_intercept_errors on;
+    error_page 307 = @transparent_proxy_no_auth;
```
Comment (alxndrsn) [inline files/nginx/odk.conf.template:75]: this is triplicated; remove 2 copies!

File: files/nginx/odk.conf.template [hunk 2]
```diff
@@ -58,6 +68,22 @@ server {
...
+  }
+
+  location @transparent_proxy_no_auth {
```
Comment (alxndrsn) [inline files/nginx/odk.conf.template:78]: Is `_no_auth` suffix relevant here?

File: service.dockerfile [hunk 3]
```diff
@@ -17,7 +17,7 @@ RUN echo "deb http://apt.postgresql.org/pub/repos/apt/ $(grep -oP 'VERSION_CODEN
 
 
 
-FROM node:${node_version}-slim as intermediate
+FROM node:${node_version}-slim AS intermediate
```
Comment (alxndrsn) [inline service.dockerfile:20]: already merged to `next`!

File: .github/workflows/test-nginx.yml [hunk 4]
```diff
@@ -15,3 +15,12 @@ jobs:
...
     - run: cd test && ./run-tests.sh
+
+    - if: always()
+      run: docker logs test-nginx-1
+    - if: always()
+      run: docker logs test-service-1 
+    - if: always()
+      run: docker logs test-enketo-1
+    - if: always()
+      run: docker logs test-mock_s3-1
```
Comment (alxndrsn) [inline .github/workflows/test-nginx.yml:26]: Move to #715

File: test/mock-http-server/index.js [hunk 5]
```diff
@@ -29,7 +38,7 @@ function withStdLogging(fn) {
   };
 }
 
-function logRequestType(method) {
+function ok(method) {
```
Comment (alxndrsn) [inline test/mock-http-server/index.js:41]: Move to #715

**Comment by lognaturel** [comment]
Capturing some quick initial thoughts that I will chew on for a bit:
- the behavior is nice. It doesn't fully hide the storage used because some headers may still come through from the provider but it does keep things like the bucket name private (though that's not actually private info)
- there's non-trivial overhead (see below)
- it would be nice for the DNS server to be configurable
- maybe the whole behavior could/should be opt-in


I did a few rounds of looking at request time and got consistent results that look like about 200ms extra than without which is itself already more than when the blobs were in the db (infrastructure-dependent, etc, etc):

Without:
Screenshot 2024-09-20 at 12 51 21 PM

With:
Screenshot 2024-09-20 at 12 55 25 PM

From db:
Screenshot 2024-09-20 at 1 02 23 PM

**Comment by alxndrsn** [comment]
Closing after discussion with @lognaturel.

**Pull request closed without merging**